rewrite-java-test/src/test/java/org/openrewrite/java/JavaTemplateTest.java

```diff
@@ -428,6 +428,93 @@ void foo() {
         );
     }
 
+    @Test
+    void replaceConstructorWithStaticFactoryMethod() {
+        rewriteRun(recipeSpec -> {
+              //everything works fine when skipping type validation
+              //recipeSpec.afterTypeValidationOptions(TypeValidation.none());
+              recipeSpec.recipe(toRecipe(() -> new JavaVisitor<>() {
+                    final MethodMatcher FactoryMethod = new MethodMatcher("f.HelloFactory createHello(String, long)");
+
+                    @Override
+                    public J visitNewClass(J.NewClass newClass, ExecutionContext executionContext) {
+                        final var result = (J.NewClass) super.visitNewClass(newClass, executionContext);
+                        final Cursor parent = getCursor().getParentOrThrow();
+                        final J.MethodDeclaration methodDeclaration = parent.firstEnclosing(J.MethodDeclaration.class);
+                        final J.ClassDeclaration classDeclaration = parent.firstEnclosing(J.ClassDeclaration.class);
+                        final var isInstantiationInFactoryMethod = FactoryMethod.matches(methodDeclaration, classDeclaration);
+                        if (isInstantiationInFactoryMethod) {
+                            return result;
+                        }
+
+                        final var template = JavaTemplate.builder("createHello(#{any()}, #{any()})")
+                          .doBeforeParseTemplate(System.out::println)
+                          .staticImports("f.HelloFactory.createHello")
+                          .build();
+
+                        maybeRemoveImport("a.Hello");
+                        maybeAddImport("f.HelloFactory", "createHello", false);
+
+                        final var arguments = result.getArguments();
+                        final var methodInvocation = template.apply(getCursor(), result.getCoordinates().replace(), arguments.get(0), arguments.get(1));
+                        if (null != ((J.MethodInvocation) methodInvocation).getMethodType()) {
+                            throw new RuntimeException("JavaTemplate does not attach the MethodType. IMO, it should.");
+                        }
```

Comment:
The conditional there should likely be `==`; but even then this check is not necessary: there's an implied check for missing types that runs with every unit test through `rewriteRun`.
```suggestion
```